### PR TITLE
ARROW-11783: [Rust] Proposal for RFCs in Rust Arrow

### PR DESCRIPTION
**Pull request by elferherrera**
The PR proposes a "light" version of the RFC process followed by the Rust community to manage the introduction of new features. The RFC objective is to be a roadmap of the desired features that people may want in the Rust language.

As an example I took the liberty of using @jorgecarleitao proposal as the first RFC that could be integrated to this folder. I also kept Jorge's layout, which I think would work better for Arrow. The template suggested by the Rust team had more details that may not be relevant to us. Let me know what you think?

Another thing that is missing is a pipeline to create the mdbook and to copy the generated files to a server. If someone knows how we can host the book in apache's server, I would appreciate your comments.

### CODE REVIEW

**Comment by andygrove** [comment]
I think this is a great way to share information about new features in one place rather than relying on mailing list discussions, JIRAs, and Google docs. Thank you for working on this @elferherrera

**Comment by paddyhoran** [comment]
@elferherrera this is a great idea.  As someone finding it hard to spend any time on Arrow but trying to keep "in the loop" this is really valuable.  Thank you.

**Comment by elferherrera** [comment]
I've been looking for ways to link the mdbook that could be generated for the RFCs to the apache arrow website and to be honest, I don't think it is possible. Also, I think it will become very cumbersome when people want to add a new RFC to the repo if we keep the RFCs within the Arrow repository.

I propose that we create another repo in the main Apache group and from there we can work with the RFCs. Something similar to what they have done to the [Arrow website](https://github.com/apache/arrow-site). I also think that by keeping it as a separate repository it will make it easier to add and maintain all the RFCs that may be received.

What do you think? @andygrove @alamb @jorgecarleitao @nevi-me

**Review by nevi-me — COMMENTED** [review]
First pass, I'll think about the process; and revert back.

File: rust/rfcs/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
```
[the remark this answers is not in the document]
Comment (elferherrera) [reply rust/rfcs/0000-template.md:1]: What is the ASF header?

File: rust/rfcs/0000-template.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Proposed changes
+[proposed-changes]: #proposed-changes
+
+Explain the proposed changes and how they can be implemented
+
+# Expected advantages
+[expected advantages]: #expected-advantages
+
+Write down a small description of the expected advantages from the proposed
+changes
```
[the remark this answers is not in the document]
Comment (elferherrera) [reply rust/rfcs/0000-template.md:27]: I think this is a good idea

File: rust/rfcs/LICENSE-APACHE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
[the remark this answers is not in the document]
Comment (elferherrera) [reply rust/rfcs/LICENSE-APACHE:1]: I dont think so. If we leave it in this repo I dont think it needs it, but Im not expert with licences

File: rust/rfcs/0000-template.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dandandan) [reply rust/rfcs/0000-template.md:1]: The Apache license text on top of each file

**Comment by andygrove** [comment]
> I've been looking for ways to link the mdbook that could be generated for the RFCs to the apache arrow website and to be honest, I don't think it is possible. Also, I think it will become very cumbersome when people want to add a new RFC to the repo if we keep the RFCs within the Arrow repository.
> 
> I propose that we create another repo in the main Apache group and from there we can work with the RFCs. Something similar to what they have done to the [Arrow website](https://github.com/apache/arrow-site). I also think that by keeping it as a separate repository it will make it easier to add and maintain all the RFCs that may be received.
> 
> What do you think? @andygrove @alamb @jorgecarleitao @nevi-me

Personally, I think we should have the RFCs in the main repo. We already publish documentation from the main repo to the Arrow web site (for example, the content under `docs/source`). Sorry, I haven't been following this closely so need to catch up on the discussions as to why you are running into issues with publishing the mdbook. I'll try and do that over the next couple days.

**Comment by emkornfield** [comment]
Stumbled across this while searching for something else.  I think this is a great idea.  From a governance point of view I would suggest the following. 
1.  Hold a vote on adopting this process for Rust on the ML.
2. Send links to ML when the a new RFC PR is made (ask that discussion happen on the PR).
3. I think RFCs should be officially approved via a vote on the ML (but I might be imagining RFCs as a much heavier wait item).

**Comment by alamb** [comment]
The Apache Arrow Rust community is moving the Rust implementation into its own dedicated github repositories [arrow-rs](https://github.com/apache/arrow-rs) and [arrow-datafusion](https://github.com/apache/arrow-datafusion). It is likely we will not merge this PR into this repository

Please see the [mailing-list](https://lists.apache.org/thread.html/rce7c61cb3f703367dc00984530016e3fcb828e5a8035655fbcccf862%40%3Cdev.arrow.apache.org%3E) thread for more details

We expect the process to take a few days and will follow up with a migration plan for the in-flight PRs.

**Pull request closed without merging**